### PR TITLE
Docker images cleanup + unttests

### PR DESCRIPTION
**Pull request by dmonakhov**
Hi,

I try to use this repo for a very first time and it is appeared very useful source of knowledge. But some recipes are broken and fail to build. This is likely because there are no easy way to test changes so I've added trivial unittest which validate that dockerfiles are healthy.  Ideally this unittests should be executed as CI scripts on commit.

List of changes
 - Add unittest to validate dockerimages
 - Update images to use most recent libraries, EFA-installer-1.28.0, aws_ofi_nccl-1.7.3-aws
 - Fix broken docker images.
 - Add hpc-benchmarks guide to run xhpl on AWS platforms.

### CODE REVIEW

**Comment by perifaws** [comment]
Unsure if we should have the HPL here. I'll let the team weight on this.

**Comment by mhuguesaws** [comment]
Why should we include a HPC benchmark in a quickstart for Distributed Machine Learning?

File: 3.test_cases/1.megatron-lm/2.distributed-training.Dockerfile [hunk 1]
```diff
@@ -66,15 +64,8 @@ RUN cd $HOME \
     && curl -O https://efa-installer.amazonaws.com/aws-efa-installer-${EFA_INSTALLER_VERSION}.tar.gz \
     && tar -xf $HOME/aws-efa-installer-${EFA_INSTALLER_VERSION}.tar.gz \
     && cd aws-efa-installer \
-    && ./efa_installer.sh -y --skip-kmod
+    && ./efa_installer.sh -y --skip-kmod --no-verify
 
-###################################################
-## Install NCCL
```
Comment (sean-smith) [inline 3.test_cases/1.megatron-lm/2.distributed-training.Dockerfile:72]: Why are we removing this?

**Comment by sean-smith** [comment]
Overall comments:
* This PR is way too much, please split it out into individual PR's based on the model
* Please comment on why we're removing NCCL from the Megatron-LM example

**Comment by sean-smith** [comment]
Also +1 on **not** adding HPL. Doesn't seem relevant to our customers.

**Comment by perifaws** [comment]
> Also +1 on **not** adding HPL. Doesn't seem relevant to our customers.

I've contacted the author, no response yet but given that this is a good PR we may pull the commits apart and resubmit (while keeping original authorship) to integrate it. @sean-smith @mhuguesaws  fine with that?

File: README.md [hunk 2]
```diff
@@ -50,7 +50,19 @@ All test cases are under `3.test_cases/`. You can go in each test case directory
...
+```
+pytest .
+```
+
+Alternatively you can run tests with out capturing stdout and keeping all docker images an other artifacts.
+```
+pytest -s --keep-artifacts=t
+```
+
+## 6. Contributors
```
[the remark this answers is not in the document]
Comment (dmonakhov) [reply README.md:65]: done

**Comment by dmonakhov** [comment]
Please review version-2

Changes  since v1:
- rename test files, so all they all now has unique names, otherwise pytest complains.
- add llama2 to nemo example
- Add missed version fixed to nemo/readme

In request to whether it is reasonable to have HPL in this repo or not. In my experience nccl-test and HPL are two must have  cluster tests before any ML workload start. It gives a confidence that HW is sane. HPL workload is very similar to pre-training workload, you can verify it via https://github.com/dmonakhov/gpu_toolbox/tree/main/ml_sample. 


For example HPCG, another test from  hpc-benchmakrs is absolutely useless from testing point of view, that is why I do not add it here

**Comment by perifaws** [comment]
> Please review version-2
> 
> Changes since v1:
> 
>     * rename test files, so all they all now has unique names, otherwise pytest complains.
> 
>     * add llama2 to nemo example
> 
>     * Add missed version fixed to nemo/readme
> 
> 
> In request to whether it is reasonable to have HPL in this repo or not. In my experience nccl-test and HPL are two must have cluster tests before any ML workload start. It gives a confidence that HW is sane. HPL workload is very similar to pre-training workload, you can verify it via https://github.com/dmonakhov/gpu_toolbox/tree/main/ml_sample.
> 
> For example HPCG, another test from hpc-benchmakrs is absolutely useless from testing point of view, that is why I do not add it here

Noted, thanks Dmitry. Let's remove the HPL for now and reserve it for another PR. That'll help merge the other contributions immediately.

**Comment by dmonakhov** [comment]
Ok. Agree, let's move hpc-benchmark out from this PR, and merge the rest.

**Review by perifaws — APPROVED** [review]
Approved, thanks for this great contribution! 🙇

**Pull request merged**